### DISCUSSION TITLE
Allow running migrations with different database user with escalated priveledges

### DISCUSSION THREAD

**Discussion by va3093** [hasura/graphql-engine#8861] [General]
In some of our migrations, we create extensions which require the postgres user to have superuser permissions. When running the migrations it seems like hasura uses the user configured with the hasura instance. We have given this user only the permissions it needs to perform its actions as an application server (ie. no super user permissions)

The problem we have is that in order to run our migrations we need to elevate the role of our hasura db user to be a super user.

### Proposed solution
I propose that the hasura cli provides arguments that allow you to specify the username and password of the db user you would like to use in order to run the migrations

### Describe alternatives you've considered

- Elevate the permissions of our application db user 😢 

### If the feature is approved, would you be willing to submit a PR?

- I don't know haskel 🙈

**Comment by seanparkross** [comment]
Converted to discussion to facilitate.

**Comment by suryarajendhran** [comment]
Is this possible in Hasura now? Our use case is to set different statement timeouts for migrations and regular queries/actions. Also open to workarounds if it'll be hard to land this as a feature.

**Reply by suryarajendhran** [reply]
@seanparkross
